corehq/apps/export/models/new.py

```diff
@@ -476,6 +503,7 @@ def __insert_system_properties(cls, table, properties, top=True):
 
         for static_column in properties:
             index, existing_column = table.get_column(static_column.item.path, static_column.item.transform)
+            (existing_column or static_column).extra_data(**data)
```

Comment:
Yeah, this function feels a bit unintuitive to me as well. What if we did something like:

    
    # add update_domain() method to StockExportColumn
    # add update_nested_repeat_count() to RowNumberColumn

    column = existing_column or copy(static_column)
    try:
        column.update_domain(domain)
    except AttributeError:
        pass
    try:
        column.update_nested_repeat_count(nested_repeat_count)
    except AttributeError:
        pass

This code is obviously less concise, but I think it's probably easier to figure out what it does at first glance. Using `isinstance()` would also be an alternative to the try/except that might look a little cleaner.
    
(I also think `extra_initialization_data()` is modifying the static columns here, so we should probably copy them before adding them to the table)